ui/components/app/snaps/snap-ui-selector/snap-ui-selector.tsx

```diff
@@ -118,15 +143,16 @@ export const SnapUISelector: React.FunctionComponent<SnapUISelectorProps> = ({
 
   const handleModalClose = () => setIsModalOpen(false);
 
-  const handleSelect = (value: string) => {
+  const handleSelect = (value: State) => {
     setSelectedOption(value);
+    onSelect?.(value);
```

Comment:
This allows to trigger an action when something is selected (for e.g. switch the selected account in the client)